sklearn/discriminant_analysis.py

```diff
@@ -641,12 +654,13 @@ def transform(self, X):
                 "transform not implemented for 'lsqr' solver (use 'svd' or 'eigen')."
             )
         check_is_fitted(self)
-
+        xp, _ = get_namespace(X)
         X = self._validate_data(X, reset=False)
+
         if self.solver == "svd":
-            X_new = np.dot(X - self.xbar_, self.scalings_)
+            X_new = (X - self.xbar_) @ self.scalings_
```

Comment:
what happens if we train using one array type (e.g. cupy), and predict using another type, e.g. numpy ndarray?

The internal state of the estimator now seems to depend on the type of the input, do we want that or should we be consistent regarding the estimator state?

Reply:
We will probably need some explicit API to convert an estimators fitted with a given  Array API namespace to be able to predict with another:

```python
from sklearn.utils import convert_to_namespace

est_cupy = Estimator().fit(X_train_cupy, y_train_cupy)

est_numpy = convert_to_namespace(est_cupy, "numpy.array_api")
est_numpy.predict(X_test_numpy)
```

but we can handle that in a later PR.

Reply:
It's now documented as part of this PR itself (with a private/experimental function for now).

Reply:
https://output.circle-artifacts.com/output/job/590b799b-29e4-46b7-9141-ae0ddd2dc6bb/artifacts/0/doc/modules/array_api.html#example-usage